bazel/foreign_cc/BUILD

```diff
@@ -91,14 +92,14 @@ envoy_cmake_external(
         "//bazel:apple": ["-lresolv"],
         "//conditions:default": [],
     }),
-    postfix_script = select({
-        "//bazel:windows_x86_64": "cp -L $EXT_BUILD_ROOT/external/com_github_c_ares_c_ares/src/lib/nameser.h $INSTALLDIR/include/nameser.h && cp -L $EXT_BUILD_ROOT/external/com_github_c_ares_c_ares/include/ares_dns.h $INSTALLDIR/include/ares_dns.h",
-        "//conditions:default": "cp -L $EXT_BUILD_ROOT/external/com_github_c_ares_c_ares/include/ares_dns.h $INSTALLDIR/include/ares_dns.h",
-    }),
-    static_libraries = select({
+    out_static_libs = select({
         "//bazel:windows_x86_64": ["cares.lib"],
         "//conditions:default": ["libcares.a"],
     }),
+    postfix_script = select({
+        "//bazel:windows_x86_64": "cp -L $EXT_BUILD_ROOT/external/com_github_c_ares_c_ares/src/lib/nameser.h $INSTALLDIR/include/nameser.h && cp -L $EXT_BUILD_ROOT/external/com_github_c_ares_c_ares/include/ares_dns.h $INSTALLDIR/include/ares_dns.h",
+        "//conditions:default": "rm -f $INSTALLDIR/include/ares_dns.h; cp -Lf $EXT_BUILD_ROOT/external/com_github_c_ares_c_ares/include/ares_dns.h $INSTALLDIR/include/ares_dns.h",
```

Comment:
note there is a change here with this `rm -f` first because I was seeing this:

```
cp: /private/var/tmp/_bazel_ksmiley/4096f01a2eae006dd7eb0d708a9f3935/sandbox/darwin-sandbox/1854/execroot/envoy/bazel-out/darwin_arm64-fastbuild/bin/external/envoy/bazel/foreign_cc/ares/include/ares_dns.h and /private/var/tmp/_bazel_ksmiley/4096f01a2eae006dd7eb0d708a9f3935/sandbox/darwin-sandbox/1854/execroot/envoy/external/com_github_c_ares_c_ares/include/ares_dns.h are identical (not copied).
```

maybe this isn't needed on macOS or with the new version of rules_foreign_cc? either way it seems safe for now